power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -939,6 +939,21 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
             });
     }
 
+    // output fault
+    template <bool sym, class Component, class ResIt>
+    std::enable_if_t<
+        std::is_base_of_v<std::forward_iterator_tag, typename std::iterator_traits<ResIt>::iterator_category> &&
+            std::is_same_v<Fault, Component>,
+        ResIt>
+    output_result(std::vector<MathOutput<sym>> const& /* math_output */, ResIt res_it) {
+        assert(construction_complete_);
+        return std::transform(components_.template citer<Component>().begin(),
+                              components_.template citer<Component>().end(), comp_coup_->fault.cbegin(), res_it,
+                              [](Fault const& fault, Idx2D /* math_id */) {
+                                  return fault.get_output();
```

Comment:
Fault would have a short circuit output right? And that output will only be available for short circuit calculation.

Reply:
yes but this is just to make it compile. Modifying the main model will be addressed in [DNR-1742](https://alliander.atlassian.net/browse/DNR-1742)

Reply:
oh sorry i see what you mean. This is just for `MathOutput`, not `ShortCircuitMathOutput`

Reply:
`ShortCircuitMathOutput` is not supported yet by the main model (cfr. [DNR-1742](https://alliander.atlassian.net/browse/DNR-1742)). therefore, this is out of scope. resolving this thread